compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/CoroutineCodegen.kt

```diff
@@ -142,16 +141,19 @@ private fun IrFunction.isStaticInlineClassReplacementDelegatingCall(): Boolean =
             parentAsClass.declarations.find { it is IrAttributeContainer && it.attributeOwnerId == attributeOwnerId && it !== this }
                 ?.isStaticInlineClassReplacement == true
 
-internal fun IrFunction.shouldContainSuspendMarkers(): Boolean = !isInvokeSuspendOfContinuation() &&
-        // These are tail-call bridges and do not require any bytecode modifications.
-        origin != IrDeclarationOrigin.FUNCTION_FOR_DEFAULT_PARAMETER &&
-        origin != JvmLoweredDeclarationOrigin.JVM_OVERLOADS_WRAPPER &&
-        origin != JvmLoweredDeclarationOrigin.SYNTHETIC_ACCESSOR &&
-        origin != JvmLoweredDeclarationOrigin.SYNTHETIC_ACCESSOR_FOR_HIDDEN_CONSTRUCTOR &&
-        origin != JvmLoweredDeclarationOrigin.SUPER_INTERFACE_METHOD_BRIDGE &&
-        origin != IrDeclarationOrigin.BRIDGE &&
-        origin != IrDeclarationOrigin.BRIDGE_SPECIAL &&
+internal val BRIDGE_ORIGINS = setOf(
+    IrDeclarationOrigin.FUNCTION_FOR_DEFAULT_PARAMETER,
+    JvmLoweredDeclarationOrigin.JVM_OVERLOADS_WRAPPER,
+    JvmLoweredDeclarationOrigin.SYNTHETIC_ACCESSOR,
+    JvmLoweredDeclarationOrigin.SYNTHETIC_ACCESSOR_FOR_HIDDEN_CONSTRUCTOR,
+    JvmLoweredDeclarationOrigin.SUPER_INTERFACE_METHOD_BRIDGE,
+    IrDeclarationOrigin.BRIDGE,
+    IrDeclarationOrigin.BRIDGE_SPECIAL,
+)
+
+internal fun IrFunction.shouldContainSuspendMarkers(): Boolean = origin !in BRIDGE_ORIGINS &&
         origin != IrDeclarationOrigin.DELEGATED_MEMBER &&
+        !isInvokeSuspendOfContinuation() &&
         !isMultifileBridge() &&
```

Comment:
`isMultifileBridge()` and `isBridgeToSuspendImplMethod()` might need to be covered too - in general, if `!shouldContainSuspendMarkers()`, then either the return value should be propagated with no boxing/unboxing, or `mv.generateCoroutineSuspendedCheck` should be done in ExpressionCodegen right after the call.

Reply:
yeah, multifile bridges try to box the result without checking for COROUTINE_SUSPENDED too... `suspendImpl` seems to be correct somehow, though.